drivers/bridge/setup_verify.go

```diff
@@ -27,11 +29,14 @@ func setupVerifyAndReconcile(config *networkConfiguration, i *bridgeInterface) e
 		return (*IPv6AddrNoMatchError)(bridgeIPv6)
 	}
 
-	// By this time we have either configured a new bridge with an IP address
-	// or made sure an existing bridge's IP matches the configuration
-	// Now is the time to cache these states in the bridgeInterface.
-	i.bridgeIPv4 = addrv4.IPNet
-	i.bridgeIPv6 = bridgeIPv6
```

Comment:
Because it is resetting the i.bridgeIPv6 to the link local IPv6 address even libnetwork passed down a global gateway address.

Actually this resetting is from the original code. What we have changed is to always execute `setupBridgeIPv4()`, even if the interface is there making this code unnecessary:

```
// Even if a bridge exists try to setup IPv4.
bridgeSetup.queueStep(setupBridgeIPv4)
```

In case ipv6 is disabled we will not program the link local anyway (via `setupBridgeIPv6()`) therefore it is fine not to execute the code I am removing.